lib/msf/ui/console/command_dispatcher/common.rb

```diff
@@ -98,6 +98,63 @@ def set_rhosts_from_addrs(rhosts)
     print_line
   end
 
+  def set_creds_from_database(publics, privates, realms, multi_creds)
+    if publics.nil? || privates.nil?
+      print_status("Unable to fill: Invalid Credentials. Check for creds index and credential options to be filled")
+      return
+    end
+    mydatastore = active_module.datastore
+    user_options = ['USERNAME', 'HttpUsername', 'SMBUser', 'DBUSER', 'FTPUSER', 'SMTPUSERNAME', 'NCSUSER', 'IMAPUSER', 'GitUsername', 'IAX_USER', 'POP2USER']
+    pass_options = ['PASSWORD', 'HttpPassword', 'SMBPass', 'DBPASS', 'FTPPASS', 'SMTPPASSWORD', 'NCSPASS', 'IMAPPASS', 'GitPassword', 'IAX_PASS', 'POP2PASS']
+    domain_options = ['DOMAIN', 'SMBDomain', 'DOMAINNAME']
+
+    # options to check if present for active module
+    if multi_creds == 1
+      if mydatastore.options.include?('USER_FILE') && mydatastore.options.include?('PASS_FILE')
+        user_file = Rex::Quickfile.new("creds-user-file")
+        mydatastore['USER_FILE'] = user_file.path
+        user_file.write(publics.join("\n")+"\n")
+        user_file.close
+        pass_file = Rex::Quickfile.new("creds-pass-file")
+        mydatastore['PASS_FILE'] = pass_file.path
+        pass_file.write(publics.join("\n")+"\n")
+        pass_file.close
+        if mydatastore.options.include?('USERPASS_FILE')
+          i = 0
+          userpass_file = Rex::Quickfile.new("creds-userpass-file")
+          mydatastore['USERPASS_FILE'] = 'file:'+userpass_file.path
```

Comment:
Does the `file:` need to be prepended to the `userpass_file.path`? It seems like it might be causing issues when the datastore option is being accessed: 
```
msf6 auxiliary(scanner/http/wordpress_login_enum) > run

[-] Msf::OptionValidateError The following options failed to validate: USERPASS_FILE
msf6 auxiliary(scanner/http/wordpress_login_enum) > options

Module options (auxiliary/scanner/http/wordpress_login_enum):

   Name                 Current Setting                                     Required  Description
   ----                 ---------------                                     --------  -----------
   BLANK_PASSWORDS      false                                               no        Try blank passwords for all users
   BRUTEFORCE           true                                                yes       Perform brute force authentication
   BRUTEFORCE_SPEED     5                                                   yes       How fast to bruteforce, from 0 to 5
   DB_ALL_CREDS         false                                               no        Try each user/password couple stored in the current database
   DB_ALL_PASS          false                                               no        Add all passwords in the current database to the list
   DB_ALL_USERS         false                                               no        Add all users in the current database to the list
   DB_SKIP_EXISTING     none                                                no        Skip existing credentials stored in the current database (Accepted: none, user, user&realm)
   ENUMERATE_USERNAMES  true                                                yes       Enumerate usernames
   PASSWORD                                                                 no        A specific password to authenticate with
   PASS_FILE            /tmp/creds-pass-file20240709-15690-9rb2kk           no        File containing passwords, one per line
   Proxies                                                                  no        A proxy chain of format type:host:port[,type:host:port][...]
   RANGE_END            10                                                  no        Last user id to enumerate
   RANGE_START          1                                                   no        First user id to enumerate
   RHOSTS               172.16.199.1                                        yes       The target host(s), see https://github.com/rapid7/metasploit-framework/wiki/Using-Metasploit
   RPORT                8000                                                yes       The target port (TCP)
   SSL                  false                                               no        Negotiate SSL/TLS for outgoing connections
   STOP_ON_SUCCESS      false                                               yes       Stop guessing when a credential works for a host
   TARGETURI            /                                                   yes       The base path to the wordpress application
   THREADS              1                                                   yes       The number of concurrent threads (max one per host)
   USERNAME                                                                 no        A specific username to authenticate as
   USERPASS_FILE        file:/tmp/creds-userpass-file20240709-15690-t0zew3  no        File containing users and passwords separated by space, one pair per line
   USER_AS_PASS         false                                               no        Try the username as the password for all users
   USER_FILE            /tmp/creds-user-file20240709-15690-9khaim           no        File containing usernames, one per line
   VALIDATE_USERS       true                                                yes       Validate usernames
   VERBOSE              true                                                yes       Whether to print output for all attempts
   VHOST                                                                    no        HTTP server virtual host


View the full module info with the info, or info -d command.
```